src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -283,6 +318,7 @@ public class ColumnFamilyStore implements ColumnFamilyStoreMBean
 
     @VisibleForTesting
     final DiskBoundaryManager diskBoundaryManager = new DiskBoundaryManager();
+    ShardBoundaries cachedShardBoundaries = null;
```

Comment:
I guess I'll save any worry about `cachedShardBoundaries` thread-safety until we get to CEP-19 and how it's used is clearer.